agent_app_sample_code/utils/load_uc_volume_to_delta_table.py

```diff
@@ -0,0 +1,182 @@
+import os
+from IPython.display import display_markdown
+import warnings
+import pyspark.sql.functions as func
+from pyspark.sql.types import StructType
+
+
+def parse_and_extract(
+    raw_doc_contents_bytes: bytes,
+    modification_time: datetime,
+    doc_bytes_length: int,
+    doc_path: str,
+    parse_file_udf: Callable[[[dict, Any]], str],
+) -> Dict[str, Any]:
+    """Parses raw bytes & extract metadata."""
+
+    try:
+        # Run the parser
+        parser_output_dict = parse_file_udf(
+            raw_doc_contents_bytes=raw_doc_contents_bytes,
+            doc_path=doc_path,
+            modification_time=modification_time,
+            doc_bytes_length=doc_bytes_length,
+        )
+
+        if parser_output_dict.get("parser_status") == "SUCCESS":
+            return parser_output_dict
+        else:
+            raise Exception(parser_output_dict.get("parser_status"))
+
+    except Exception as e:
+        status = f"An error occurred: {e}\n{traceback.format_exc()}"
+        warnings.warn(status)
+        return {
+            "doc_content": "",
+            "doc_uri": "",
+            "parser_status": status,
+        }
+
+
+def get_parser_udf(
+    # extract_metadata_udf: Callable[[[dict, Any]], str],
+    parse_file_udf: Callable[[[dict, Any]], str],
+    spark_dataframe_schema: StructType,
+):
+    """Gets the Spark UDF which will parse the files in parallel.
+
+    Arguments:
+      - extract_metadata_udf: A function that takes parsed content and extracts the metadata
+      - parse_file_udf: A function that takes the raw file and returns the parsed text.
+      - spark_dataframe_schema: The resulting schema of the document delta table
+    """
+    # This UDF will load each file, parse the doc, and extract metadata.
+    parser_udf = func.udf(
+        lambda raw_doc_contents_bytes, modification_time, doc_bytes_length, doc_path: parse_and_extract(
+            raw_doc_contents_bytes,
+            modification_time,
+            doc_bytes_length,
+            doc_path,
+            parse_file_udf,
+        ),
+        returnType=spark_dataframe_schema,
+    )
+    return parser_udf
+
+
+def load_uc_volume_to_delta_table(
+    source_path: str,
+    dest_table_name: str,
+    parse_file_udf: Callable[[[dict, Any]], str],
+    spark_dataframe_schema: StructType
+) -> str:
+    if not os.path.exists(source_path):
+        raise ValueError(
+            f"{source_path} passed to `load_uc_volume_to_delta_table` does not exist."
+        )
+
+    # Load the raw riles
+    raw_files_df = (
+        spark.read.format("binaryFile").option("recursiveFileLookup", "true")
+        .load(source_path)
+    )
+
+    # Check that files were present and loaded
+    if raw_files_df.count() == 0:
+        display(
+            f"`{source_path}` does not contain any files.  Open the volume and upload at least file."
+        )
+        raise Exception(f"`{source_path}` does not contain any files.")
```

Comment:
Maybe just raise the exception here? Curious why we have the `display()` call too